prov/rxm/src/rxm_init.c

```diff
@@ -237,7 +240,8 @@ static void rxm_alter_info(const struct fi_info *hints, struct fi_info *info)
 			}
 
 			if (!hints->domain_attr ||
-			    hints->domain_attr->data_progress != FI_PROGRESS_AUTO)
+			    hints->domain_attr->data_progress != FI_PROGRESS_AUTO ||
```

Comment:
Personal opinion: If we cannot support auto progress, we should reject the info call with the appropriate provider not found if fi_progress_auto was explicitly requested. While applications are supposed to check the returned fi_infi structure, it feels like most users don't.

Reply:
That was my original thought, and might be the best solution. What made me lean this way was from the fi_getinfo MAN page:

domain_attr - domain attributes
Optionally supplied domain attributes. Domain attributes may be specified and returned as part of fi_getinfo. When provided as hints, requested values of struct fi_domain_attr should be set. On output, the actual domain attributes that can be provided will be returned. Output values will be greater than or equal to requested input values. See fi_domain(3) for details.

Since FI_MANUAL_PROGRESS = 2 and FI_AUTO_PROGRESS=1. However, I agree with your point, that I doubt it will be checked.

Reply:
I am going to update the PR to not return the fi_info if data FI_AUTO_PROGRESS and FI_ATOMIC are requested.

Reply:
@jswaro - Updated to not return fi_info if FI_ATOMIC and data FI_PROGRESS_AUTO is requested. That definitely is cleaner and I believe is the correct way to handle it. Thanks!